packages/lib-classifier/src/plugins/tasks/DrawingTask/models/markings/Mark.js

```diff
@@ -5,5 +5,10 @@ const Mark = types.model('Mark', {
   frame: types.optional(types.number, 0),
   toolIndex: types.optional(types.number, 0)
 })
+  .views(self => ({
+    get isValid () {
+      return true
```

Comment:
Should a mark be valid by default? If you forget to write a new `isValid` method when writing a new tool, you could create invalid marks without meaning to.

Reply:
If this does change, it's probably worth adding a console message to say the default `isValid` method has kicked in

Reply:
You'd notice when testing, I'd hope, because valid marks won't be deleted from the drawing canvas.

Reply:
Points are an example of a type of mark that doesn't have a validation.

Reply:
`isValid` is optional, not required.

Reply:
If you forget to override `isValid` when making a new tool, invalid marks won't be deleted either, right? Might be an edge case, and you'd hope it'd get caught in review, but it might not.

Also, for marks that end up out of bounds, `isValid` might be a sensible way to handle them. You could have an `isOutOfBounds` view that checks whether any of the coords constituting a mark is outside of the drawing area, and `isValid` calls that as part of its check.

Reply:
>If you forget to override isValid when making a new tool, invalid marks won't be deleted either, right? Might be an edge case, and you'd hope it'd get caught in review, but it might not.

Exactly, and you'll notice that when they don't vanish from the canvas in testing.

Out-of-bounds is out of scope for this PR, but these models don't have a concept of a drawing area. See also the calculation of the delete button position. Valid position on the drawing canvas should probably be calculated in the view, which does have the canvas bounds.

Reply:
Anyway, if you look through the original code in PFE, `isValid` is optional for marks.

Reply:
👍